--- a/third_party/FFmpeg/libavformat/options.c
+++ b/third_party/FFmpeg/libavformat/options.c
@@ -45,16 +45,23 @@ static const char* format_to_name(void* ptr)
 static void *format_child_next(void *obj, void *prev)
 {
     AVFormatContext *s = obj;
+#ifdef __APPLE__
+    if (!prev && s->priv_data &&
+        ((s->iformat && s->iformat->priv_class) ||
+         (s->oformat && s->oformat->priv_class)))
+        return s->priv_data;
+#else
     if (!prev && s->priv_data &&
         ((s->iformat && s->iformat->priv_class) ||
           s->oformat && s->oformat->priv_class))
         return s->priv_data;
+#endif
     if (s->pb && s->pb->av_class && prev != s->pb)
         return s->pb;
     return NULL;
 }
 
-#if FF_API_CHILD_CLASS_NEXT
+#if FF_API_CHILD_CLASS_NEXT && !defined(__APPLE__)
 static const AVClass *format_child_class_next(const AVClass *prev)
 {
     const AVInputFormat *ifmt = NULL;
@@ -158,7 +165,7 @@ static const AVClass av_format_context_class = {
     .option         = avformat_options,
     .version        = LIBAVUTIL_VERSION_INT,
     .child_next     = format_child_next,
-#if FF_API_CHILD_CLASS_NEXT
+#if FF_API_CHILD_CLASS_NEXT && !defined(__APPLE__)
     .child_class_next = format_child_class_next,
 #endif
     .child_class_iterate = format_child_class_iterate,
@@ -171,6 +178,15 @@ static int io_open_default(AVFormatContext *s, AVIOContext **pb,
 {
     int loglevel;
 
+#ifdef __APPLE__
+    if (!strcmp(url, s->url) ||
+        (s->iformat && !strcmp(s->iformat->name, "image2")) ||
+        (s->oformat && !strcmp(s->oformat->name, "image2"))
+    ) {
+        loglevel = AV_LOG_DEBUG;
+    } else
+        loglevel = AV_LOG_INFO;
+#else
     if (!strcmp(url, s->url) ||
         s->iformat && !strcmp(s->iformat->name, "image2") ||
         s->oformat && !strcmp(s->oformat->name, "image2")
@@ -178,6 +194,7 @@ static int io_open_default(AVFormatContext *s, AVIOContext **pb,
         loglevel = AV_LOG_DEBUG;
     } else
         loglevel = AV_LOG_INFO;
+#endif
 
     av_log(s, loglevel, "Opening \'%s\' for %s\n", url, flags & AVIO_FLAG_WRITE ? "writing" : "reading");
