--- a/third_party/llvm/include/llvm/ADT/BitVector.h
+++ b/third_party/llvm/include/llvm/ADT/BitVector.h
@@ -155,6 +155,11 @@ public:
     return !any();
   }
 
+  #ifdef __APPLE__
+  #pragma clang diagnostic push
+  #pragma clang diagnostic ignored "-Wshorten-64-to-32"
+  #endif
+
   /// find_first - Returns the index of the first set bit, -1 if none
   /// of the bits are set.
   int find_first() const {
