--- a/third_party/disruptorplus/include/disruptorplus/multi_threaded_claim_strategy.hpp
+++ b/third_party/disruptorplus/include/disruptorplus/multi_threaded_claim_strategy.hpp
@@ -312,7 +312,11 @@ namespace disruptorplus
                         return false;
                     }
                 }
+                #ifdef __APPLE__
+                reducedCount = std::min<sequence_t>(count, static_cast<sequence_t>(diff + 1));
+                #else
                 reducedCount = std::min(count, static_cast<sequence_t>(diff + 1));
+                #endif
             } while (!m_nextClaimable.compare_exchange_weak(
                 sequence,
                 static_cast<sequence_t>(sequence + reducedCount),
