--- a/third_party/FFmpeg/libavcodec/bitstream.c
+++ b/third_party/FFmpeg/libavcodec/bitstream.c
@@ -405,7 +405,11 @@ int ff_init_vlc_from_lengths(VLC *vlc_arg, int nb_bits, int nb_codes,
             else
                 sym = i;
             buf[j].symbol = sym + offset;
+#ifdef __APPLE__
+            buf[j++].code = (uint32_t)code;
+#else
             buf[j++].code = code;
+#endif
         } else if (len <  0) {
             len = -len;
         } else
