--- a/third_party/FFmpeg/libavcodec/simple_idct.c
+++ b/third_party/FFmpeg/libavcodec/simple_idct.c
@@ -75,8 +75,13 @@ static inline void idct4col_put(uint8_t *dest, ptrdiff_t line_size, const int16_
     a1 = col[8*2];
     a2 = col[8*4];
     a3 = col[8*6];
+#ifdef __APPLE__
+    c0 = ((a0 + a2) * (1 << (CN_SHIFT - 1))) + (1 << (C_SHIFT - 1));
+    c2 = ((a0 - a2) * (1 << (CN_SHIFT - 1))) + (1 << (C_SHIFT - 1));
+#else
     c0 = ((a0 + a2) * (1 << CN_SHIFT - 1)) + (1 << (C_SHIFT - 1));
     c2 = ((a0 - a2) * (1 << CN_SHIFT - 1)) + (1 << (C_SHIFT - 1));
+#endif
     c1 = a1 * C1 + a3 * C2;
     c3 = a1 * C2 - a3 * C1;
     dest[0] = av_clip_uint8((c0 + c1) >> C_SHIFT);
