--- a/third_party/FFmpeg/libavcodec/wmadec.c
+++ b/third_party/FFmpeg/libavcodec/wmadec.c
@@ -109,11 +109,17 @@ static av_cold int wma_decode_init(AVCodecContext *avctx)
     for (i = 0; i < s->nb_block_sizes; i++)
         ff_mdct_init(&s->mdct_ctx[i], s->frame_len_bits - i + 1, 1, 1.0 / 32768.0);
 
-    if (s->use_noise_coding) {
-        ff_init_vlc_from_lengths(&s->hgain_vlc, HGAINVLCBITS, FF_ARRAY_ELEMS(ff_wma_hgain_hufftab),
-                                 &ff_wma_hgain_hufftab[0][1], 2,
-                                 &ff_wma_hgain_hufftab[0][0], 2, 1, -18, 0, avctx);
-    }
+if (s->use_noise_coding) {
+#ifdef __APPLE__
+    ff_init_vlc_from_lengths(&s->hgain_vlc, HGAINVLCBITS, FF_ARRAY_ELEMS(ff_wma_hgain_hufftab),
+                             (const int8_t *)&ff_wma_hgain_hufftab[0][1], 2,
+                             (const uint8_t *)&ff_wma_hgain_hufftab[0][0], 2, 1, -18, 0, avctx);
+#else
+    ff_init_vlc_from_lengths(&s->hgain_vlc, HGAINVLCBITS, FF_ARRAY_ELEMS(ff_wma_hgain_hufftab),
+                             &ff_wma_hgain_hufftab[0][1], 2,
+                             &ff_wma_hgain_hufftab[0][0], 2, 1, -18, 0, avctx);
+#endif
+}
 
     if (s->use_exp_vlc)
         init_vlc(&s->exp_vlc, EXPVLCBITS, sizeof(ff_aac_scalefactor_bits), // FIXME move out of context
