--- a/third_party/zstd/examples/dictionary_decompression.c
+++ b/third_party/zstd/examples/dictionary_decompression.c
@@ -11,7 +11,11 @@
 
 #include <stdio.h>     // printf
 #include <stdlib.h>    // free
-#include <zstd.h>      // presumes zstd library is installed
+#ifdef __APPLE__
+#include "zstd.h"
+#else
+#include <zstd.h>
+#endif      // presumes zstd library is installed
 #include "common.h"    // Helper functions, CHECK(), and CHECK_ZSTD()
 
 /* createDict() :
