diff --git a/third_party/zstd/examples/multiple_streaming_compression.c b/third_party/zstd/examples/multiple_streaming_compression.c
index b12ad03..62d6d33 100644
--- a/third_party/zstd/examples/multiple_streaming_compression.c
+++ b/third_party/zstd/examples/multiple_streaming_compression.c
@@ -17,7 +17,11 @@
 #include <stdio.h>     // printf
 #include <stdlib.h>    // free
 #include <string.h>    // memset, strcat
-#include <zstd.h>      // presumes zstd library is installed
+#ifdef __APPLE__
+#include "zstd.h"
+#else
+#include <zstd.h>
+#endif      // presumes zstd library is installed
 #include "common.h"    // Helper functions, CHECK(), and CHECK_ZSTD()
 
 typedef struct {
