--- a/third_party/zstd/examples/streaming_compression_thread_pool.c
+++ b/third_party/zstd/examples/streaming_compression_thread_pool.c
@@ -12,7 +12,11 @@
 #include <stdio.h>     // printf
 #include <stdlib.h>    // free
 #include <string.h>    // memset, strcat, strlen
-#include <zstd.h>      // presumes zstd library is installed
+#ifdef __APPLE__
+#include "zstd.h"
+#else
+#include <zstd.h>
+#endif      // presumes zstd library is installed
 #include "common.h"    // Helper functions, CHECK(), and CHECK_ZSTD()
 #include <pthread.h>
 
