--- a/third_party/capstone/utils.c
+++ b/third_party/capstone/utils.c
@@ -188,7 +188,12 @@ void append_to_str_lower(char *str, size_t str_size, const char *src) {
 		return;
 	}
 
-	int i = dest - str;
+#ifdef __APPLE__
+    ptrdiff_t diff = dest - str;
+    int i = (int)diff;
+#else
+    int i = dest - str;
+#endif
 	for (int j = 0; (i < str_size) && (j < strlen(src)); ++i, ++j) {
 		str[i] = tolower(src[j]);
 	}

