--- a/third_party/capstone/tests/integration/compat_header/src/test_sysz_compatibility_header.c
+++ b/third_party/capstone/tests/integration/compat_header/src/test_sysz_compatibility_header.c
@@ -5,7 +5,11 @@
 #include <inttypes.h>
 
 #define CAPSTONE_SYSTEMZ_COMPAT_HEADER
+#ifdef __APPLE__
+#include "capstone/capstone.h"
+#else
 #include <capstone/capstone.h>
+#endif
 
  // 0  5a 0f 1f ff        a	%r0, 0xfff(%r15,%r1)
 	// ID: 1 (a)
