--- a/third_party/FFmpeg/libavutil/fixed_dsp.c
+++ b/third_party/FFmpeg/libavutil/fixed_dsp.c
@@ -137,7 +137,11 @@ static int scalarproduct_fixed_c(const int *v1, const int *v2, int len)
 static void butterflies_fixed_c(int *v1s, int *v2, int len)
 {
     int i;
+#ifdef __APPLE__
+    int *v1 = v1s;
+#else
     unsigned int *v1 = v1s;
+#endif
 
     for (i = 0; i < len; i++){
         int t = v1[i] - v2[i];
